src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -528,7 +595,7 @@ void ParquetBlockInputFormat::initializeIfNeeded()
     if (is_stopped)
         return;
 
-    metadata = parquet::ReadMetaData(arrow_file);
+    metadata = getFileMetaData();
```

Comment:
what will happen if the object metadata was put in the cache, but later on that object was deleted from S3?

Reply:
It'll fail on [S3::getObjectInfo call](https://github.com/ClickHouse/ClickHouse/blob/5efa76fc86cbf1ce23fb26508e55c9954a4e78f5/src/IO/S3/getObjectInfo.cpp#L88) that is made prior to reaching this code.

Reply:
```
SETTINGS input_format_parquet_use_metadata_cache = 1, input_format_parquet_filter_push_down = 1, log_comment = 'abc', remote_filesystem_read_prefetch = 0

Query id: 2d36d362-894b-4ad5-a075-6960bca9de88


Elapsed: 148.109 sec. 

Received exception from server (version 24.12.2):
Code: 499. DB::Exception: Received from localhost:9000. DB::Exception: Failed to get object info: No response body.. HTTP response code: 404: while reading adir/test0.parquet: The table structure cannot be extracted from a Parquet format file. You can specify the structure manually. (S3_ERROR)

```